tensorrt_llm/_torch/models/modeling_llama.py

```diff
@@ -53,36 +55,163 @@ def __init__(
+        mrope_config: Optional[dict] = None,
+        all_reduce_params: Optional[AllReduceParams] = None,
+        lora_params: Optional[dict] = None,
+        **kwargs,
+    ) -> torch.Tensor:
+        qkv = self.qkv_proj(hidden_states)
+        is_fused_qkv = False
+        if isinstance(self.attn, TrtllmAttention):
+            is_fused_qkv = True
+
+        if self.qk_norm is not None:
+            assert self.attn_backend == "FLASHINFER", "attention qk_norm is only supported w/ flashinfer backend"
+            # Apply rope ahead of qk_norm
+            if self.pos_embd_params is None and position_ids is not None and self.rotary_emb is not None:
+
+                q, k, v = qkv.split([self.q_size, self.kv_size, self.kv_size],
+                                    dim=-1)
+                q, k = self.rotary_emb(position_ids, [q, k], attn_metadata)
+                qkv = torch.concat([q, k, v], dim=-1)
+
+            # TODO: make this more efficient.
+            q, k, v = qkv.split([self.q_size, self.kv_size, self.kv_size],
+                                dim=-1)
+            do_multi_stream = torch.cuda.is_current_stream_capturing(
+            ) and self.aux_stream is not None
+            if do_multi_stream:
+                self.ln_events[0].record()
+                k = k.reshape(-1, self.head_dim)
+                k = self.qk_norm(k).reshape(-1, self.kv_size)
+                with torch.cuda.stream(self.aux_stream):
+                    self.ln_events[0].wait()
+                    q = q.reshape(-1, self.head_dim)
+                    q = self.qk_norm(q).reshape(-1, self.q_size)
+                    self.ln_events[1].record()
+                self.ln_events[1].wait()
+            else:
+                q = q.reshape(-1, self.head_dim)
+                k = k.reshape(-1, self.head_dim)
+                q = self.qk_norm(q).reshape(-1, self.q_size)
+                k = self.qk_norm(k).reshape(-1, self.kv_size)
+            qkv = torch.concat([q, k, v], dim=-1)
+
+        if self.attn_temperature_tuning:
+            # this must be a nope layer
+            assert position_ids is not None, "attn_temperature_tuning requires position_ids"
+            q, k, v = qkv.split([self.q_size, self.kv_size, self.kv_size],
+                                dim=-1)
+            attn_scale = self._get_attn_scale(position_ids)
+            q = (q * attn_scale).to(q.dtype)
+            qkv = torch.concat(
+                [q.contiguous(), k.contiguous(),
+                 v.contiguous()], dim=-1)
+
+        if lora_params is not None:
+            qkv_lora = self.splitted_qkv_lora(hidden_states, lora_params,
+                                              self.layer_idx)
+            if qkv_lora is not None:
+                qkv = qkv + qkv_lora
+
+            qkv_lora = self.fused_qkv_lora(hidden_states, lora_params,
+                                           self.layer_idx)
+            if qkv_lora is not None:
+                qkv = qkv + qkv_lora
```

Comment:
Let's remove these for now. I'm not even sure if they're inserted at the right place.